File: 37주차/네트워크.py
def solution(n, computers):
    answer = 0
    g = [[] for _ in range(n+1)]
    for i in range(n):
        for j in range(n):
            if i !=j and computers[i][j] == 1:
                g[i+1].append(j+1)

    visited = [0]*(n+1)
    
    def dfs(i):
        visited[i] = 1
        for v in g[i]:
            if not visited[v]:
                dfs(v)
    # def dfs(i):
    #     visited[i]= 1
    #     for j in range(1,n+1):
    #         if computers[i-1][j-1] and not visited[j]:
    #             dfs(j)
    for i in range(1,n+1):
        if not visited[i]:
            dfs(i)
            answer += 1
    return answer
# 프로그래머스 네트워크

File: 47주차/네트워크.js
function solution(n, computers) {
  var answer = 0;
  let visited = new Array(n).fill(0);
  function dfs(x) {
    visited[x] = 1;

    for (let i = 0; i < n; i++) {
      // x번 노드와 연결된 0~n-1번 노드 중에서 값이 1인것 찾기.
      if (!visited[i] && computers[x][i]) dfs(i);
    }
  }

  for (let i = 0; i < n; i++) {
    if (!visited[i]) {
      dfs(i);
      answer += 1;
    }
  }
  return answer;
}
// 노드끼리 연결여부를 나타낸 2차원 배열을 가지고 네트워크가 몇 개 인지 맞추기.
// 프로그래머스 dfs.
